src/tty-ask-password-agent/tty-ask-password-agent.c

```
@@ -510,12 +510,8 @@ static int process_and_watch_password_files(void) {
         if (notify < 0)
                 return log_error_errno(errno, "Failed to allocate directory watch: %m");
 
-        if (inotify_add_watch(notify, "/run/systemd/ask-password", IN_CLOSE_WRITE|IN_MOVED_TO) < 0) {
-                if (errno == ENOSPC)
-                        return log_error_errno(errno, "Failed to add /run/systemd/ask-password to directory watch: inotify watch limit reached");
-                else
-                        return log_error_errno(errno, "Failed to add /run/systemd/ask-password to directory watch: %m");
-        }
+        if (inotify_add_watch(notify, "/run/systemd/ask-password", IN_CLOSE_WRITE|IN_MOVED_TO) < 0)
+                return log_error_errno(errno, "Failed to watch /run/systemd/ask-password: %m");
```

This was added because the default message is very misleading: is says that the *disk* is filled, but in fact the inotify watch limit is the problem. I think it's better to keep this as is.

---

The thing is that there're a few other places that add watches but don't make any specific case when the watch limit is reached. Maybe we could add an helper then ?

---

`inotify_add_watch_and_warn` sounds good.

---

OK I'll add the helper.